### PR TITLE
update elastic-agent-libs

### PR DESCRIPTION
**Pull request by AndersonQ**
## What is the problem this PR solves?

fleet-server using an outdated version of elastic-agent-libs

## How does this PR solve the problem?

by updating `elastic-agent-libs` ot its latest version

## How to test this PR locally

Ensure mTLS is still working

adjust the IPs/hostnames as needed

### build a fleet-server out of this PR
```shell
go build
```

### you might need to build an 8.16 agent out of main:
```
AGENT_PACKAGE_VERSION=8.16.0-SNAPSHOT BEATS_VERSION=8.16.0-SNAPSHOT DEV=true SNAPSHOT=true EXTERNAL=true PLATFORMS="linux/amd64" PACKAGES="tar.gz" mage package
```

### add your fleet server built to the agent package

```shell
tar -xf elastic-agent-8.16.0-SNAPSHOT-linux-x86_64.tar.gz
cp path/tp/your/fleet-server ./elastic-agent-8.16.0-SNAPSHOT-linux-x86_64/data/elastic-agent-*/components/fleet-server
```

### create 2 TLS certificates
 - use `elastic-agent-libs/testing/certutil/cmd` to create the certificates. Make sure to use `elastic-agent-libs` with [this PR](https://github.com/elastic/elastic-agent-libs/pull/243) merged or use the PR branch
 - either make your fleet-server reachable on `fleet-server` or change `-name fleet-server` to a valid DNS for your fleet-server.

```
go run main.go -prefix server -name fleet-server -noip
go run main.go -client -prefix client
```
you should have:
```
-rw------- 1 ainsoph ainsoph  312 Oct 24 16:36 client-ca_key.pem
-rw------- 1 ainsoph ainsoph  928 Oct 24 16:36 client-ca.pem
-rw------- 1 ainsoph ainsoph  312 Oct 24 16:36 client-client_key.pem
-rw------- 1 ainsoph ainsoph  895 Oct 24 16:36 client-client.pem

-rw------- 1 ainsoph ainsoph  312 Oct 24 16:37 server-ca_key.pem
-rw------- 1 ainsoph ainsoph  932 Oct 24 16:37 server-ca.pem
-rw------- 1 ainsoph ainsoph  312 Oct 24 16:37 server-fleet-server_key.pem
-rw------- 1 ainsoph ainsoph  928 Oct 24 16:37 server-fleet-server.pem
```

### add a fleet server with mTLS

```
elastic-agent install -nf \
--url=https://fleet-server:8220 \
--fleet-server-es=https://es.elastic-cloud.com:443 \
--fleet-server-service-token=a-service-token \
--fleet-server-policy=fleet-server-policy \
--certificate-authorities=/root/certs/server-ca.pem,/root/certs/client-ca.pem,/etc/ssl/certs/ca-certificates.crt \
--fleet-server-cert=/root/certs/server-fleet-server.pem \
--fleet-server-cert-key=/root/certs/server-fleet-server_key.pem \
--elastic-agent-cert=/root/certs/client-client.pem \
--elastic-agent-cert-key=/root/certs/client-client_key.pem \
--fleet-server-client-auth=required \
--fleet-server-port=8220
```

### add an agent to that policy
```
elastic-agent install -nf \
--url=https://fleet-server:8220 \
--enrollment-token=a-token \
--certificate-authorities=/root/certs/server-ca.pem,/etc/ssl/certs/ca-certificates.crt \
--elastic-agent-cert=/root/certs/client-client.pem \
--elastic-agent-cert-key=/root/certs/client-client_key.pem 
```

## Design Checklist

- ~~[ ] I have ensured my design is stateless and will work when multiple fleet-server instances are behind a load balancer.~~
- ~~[ ] I have or intend to scale test my changes, ensuring it will work reliably with 100K+ agents connected.~~
- ~~[ ] I have included fail safe mechanisms to limit the load on fleet-server: rate limiting, circuit breakers, caching, load shedding, etc.~~

## Checklist

- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/fleet-server#changelog)

## Related issues

- N/A

### CODE REVIEW

**Comment by cmacknz** [comment]
Needs a changelog for the mTLS certificate verification change at least, and anything else significant that has come in as part of this version bump.

**Comment by AndersonQ** [comment]
the git push didn't go through, it's there now

File: changelog/fragments/1729782694-Fleet-Server-uses-'ssl.verification_mode:-certificate'-by-default-for-incomming-client-connections.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# - enhancement: extends functionality but does not break or fix existing behavior
+# - feature: new functionality
+# - known-issue: problems that we are aware of in a given version
+# - security: impacts on the security of a product or a user’s deployment.
+# - upgrade: important information for someone upgrading from a prior version
+# - other: does not fit into any of the other categories
+kind: feature
+
+# Change summary; a 80ish characters long description of the change.
+summary: Fleet Server uses 'ssl.verification_mode: certificate' by default for incomming client connections
```
Comment (cmacknz) [inline changelog/fragments/1729782694-Fleet-Server-uses-'ssl.verification_mode:-certificate'-by-default-for-incomming-client-connections.yaml:14]: I don't think this is clear enough. Fleet Server's TLS certificate is a critical part of agent security. Explain why this was done and what the possible consequences are, in particular separate out the impact for mTLS and standard TLS if they are different.

**Comment by cmacknz** [comment]
The PR description here should also reflect what the actual change in elastic-agent-libs is, because there is only one https://github.com/elastic/elastic-agent-libs/commit/4babafd5ed1e5079acf74212ed3da01740b22de7

We should also detail what testing has or needs to happen to make sure this doesn't break anything or regress TLS security accidentally.

**Review by ycombinator — APPROVED** [review]

**Pull request merged**